File: Basics/Kotlin/app/src/main/java/com/suavepirate/kotlinbasics/samples/dataClass/Dog.kt
package com.suavepirate.kotlinbasics.samples.dataClass

data class Dog(val name: String, val age: Int, val breed: String)

val dog1 = Dog("Bentley", 5, "Staffordshire Terrier")

val dog1Copy = dog1.copy()

val areSame = dog1 == dog1Copy // true

File: Basics/C#/CSharp.Basics/Samples/Encapsulation/Dog.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp.Basics.Samples.Encapsulation
{
    public class Dog
    {
        public string Name { get; set; }
        protected int _age;
        private Breed _breed;
        private class Breed
        {
            public string Name { get; set; }
        }
    }
}
